test/actions/pulp3/orchestration/file_upload_test.rb

```diff
@@ -50,14 +51,16 @@ def test_duplicate_upload
           :root_repository_id => @repo.root.id,
           :content_view_id => @repo.content_view.id)
       assert_equal repository_reference.repository_href + "versions/1/", @repo.version_href
+
       VCR.use_cassette(cassette_name + '_binary_duplicate', :match_requests_on => [:method, :path, :params]) do
-        action_result = ForemanTasks.sync_task(::Actions::Pulp3::Orchestration::Repository::UploadContent, @repo, @primary, @file, "file")
+        action_result = ForemanTasks.sync_task(::Actions::Pulp3::Orchestration::Repository::UploadContent, @repo, @primary, @file1, "file")
       end
       assert_equal "success", action_result.result
+      @repo.reload
       repository_reference = Katello::Pulp3::RepositoryReference.find_by(
           :root_repository_id => @repo.root.id,
           :content_view_id => @repo.content_view.id)
-      assert_equal repository_reference.repository_href + "versions/1/", @repo.version_href
+      assert_equal repository_reference.repository_href + "versions/2/", @repo.version_href
```

Comment:
this does validate that the repository was updated 2x, but it might be better to fetch the repository version and verify the # of files in it

Reply:
Updated to check content_count.. 👍🏽